### PR TITLE
Using tribe_asset for general Assets on Core

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#40267](http://central.tri.be/issues/40267)
---

Stopped using `Tribe__Events__Template_Factory::asset_package` in favor of `tribe_asset`
for a lot of the required Front-end and Back-end Assets for The Events Calendar Core

---

https://github.com/moderntribe/events-filterbar/pull/136
https://github.com/moderntribe/events-community/pull/306
https://github.com/moderntribe/events-pro/pull/822
https://github.com/moderntribe/tribe-common/pull/657

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:223](https://github.com/moderntribe/the-events-calendar/blob/52f6e3e45d0442529908f6cfb3a9923ce4a36e26/src/Tribe/Assets.php#L223)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:272](https://github.com/moderntribe/the-events-calendar/blob/259fa30997cf0e8d3eccca557296ed8770991e35/src/Tribe/Assets.php#L272)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:272](https://github.com/moderntribe/the-events-calendar/blob/ed1cad0b5191de074b0370ec5f2289da9a6f655a/src/Tribe/Assets.php#L272)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:272](https://github.com/moderntribe/the-events-calendar/blob/6dc9df2ccf1428726164a98d8c416d54b748b85d/src/Tribe/Assets.php#L272)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:272](https://github.com/moderntribe/the-events-calendar/blob/220d149663089765fcf9a1a4c035f465720f3128/src/Tribe/Assets.php#L272)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:268](https://github.com/moderntribe/the-events-calendar/blob/4769fa2d682f9e07c29d082a49b739199549290b/src/Tribe/Assets.php#L268)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:302](https://github.com/moderntribe/the-events-calendar/blob/1da6d7597741e9ec2338555bfe22c5f0e8e45c20/src/Tribe/Assets.php#L302)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:306](https://github.com/moderntribe/the-events-calendar/blob/edcf941c92bae5c85c047ae8d09ac07a72808c34/src/Tribe/Assets.php#L306)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Assets.php:306](https://github.com/moderntribe/the-events-calendar/blob/b5eea2f47696601b6c8ac790a37a16ac5ea6ff6d/src/Tribe/Assets.php#L306)

> via jenkins/codesniffer-phpcs

**Comment by mitogh** [comment]
Sending a heads up here on a Post type that support tickets is triggering the following error: 

https://cl.ly/1E3e1t0u2O1e

```
events-dynamic.js?ver=4.6.14:14 Uncaught ReferenceError: tribe_dynamic_help_text is not defined
    at events-dynamic.js?ver=4.6.14:14
    at events-dynamic.js?ver=4.6.14:167
(anonymous) @ events-dynamic.js?ver=4.6.14:14
(anonymous) @ events-dynamic.js?ver=4.6.14:167
```

File: src/Tribe/Bar.php [hunk 1]
```diff
@@ -37,17 +39,16 @@ public function __construct() {
...
 			return;
 		}
 
-		$show_bar_filter = in_array(
-			get_post_type(), array(
-				Tribe__Events__Main::VENUE_POST_TYPE,
-				Tribe__Events__Main::ORGANIZER_POST_TYPE,
-			)
-		) ? false : true;
+		$not_in_post_types = array(
```
Comment (mitogh) [inline src/Tribe/Bar.php]: I think this variable should be called `$allowed_types` as the `not` is applied when using the `!` operator during the comparison on `in_array`

File: src/Tribe/Main.php [hunk 2]
```diff
@@ -4772,12 +4596,152 @@ public function maybe_set_currency_position_with_post( $reverse_position, $post_
...
+		 *
+		 * @param array $post_types
+		 * @return array The filtered post types
+		 */
+		public function filter_meta_chunker_post_types( array $post_types ) {
+			$post_types = array_merge(
+				$post_types,
+				array(
+					self::POSTTYPE,
+					self::VENUE_POST_TYPE,
```
Comment (mitogh) [inline src/Tribe/Main.php]: `Tribe__Events__Venue::POSTTYPE` and `Tribe__Events__Organizer::POSTTYPE`

File: src/Tribe/Template_Factory.php [hunk 3]
```diff
@@ -538,6 +469,86 @@ public function comments_off( $option_value, $option_name ) {
...
+	 *
+	 * @deprecated TBD
+	 *
+	 * @param string $name
+	 * @param array  $deps Dependents
+	 */
+	public static function asset_package( $name, $deps = array() ) {
+
+		$common = Tribe__Events__Main::instance();
+		$prefix = 'tribe-events'; // Tribe__Events__Main::POSTTYPE;
```
Comment (mitogh) [inline src/Tribe/Template_Factory.php]: The post type is `tribe_events` tho not sure if the comment should be more explicit about that

File: src/Tribe/Template_Factory.php [hunk 4]
```diff
@@ -538,6 +469,86 @@ public function comments_off( $option_value, $option_name ) {
...
+		self::prepare_asset_package_request( $asset, $name, $deps, $vendor_url, $prefix, $tec );
+	}
+
+	/**
+	 * Retrieves the appropriate asset factory instance
+	 *
+	 * @deprecated TBD
+	 */
+	protected static function get_asset_factory_instance( $name ) {
+		$asset = Tribe__Events__Asset__Factory::instance()->make_for_name( $name );
```
Comment (mitogh) [inline src/Tribe/Template_Factory.php]:
```
return Tribe__Events__Asset__Factory::instance()->make_for_name( $name );
```

**Comment by bordoni** [comment]
Nice @mitogh will check on that.

**Comment by mitogh** [comment]
`common` module hash should be removed?

**Comment by bordoni** [comment]
Not on this case because of in QA testing.

File: src/Tribe/Template_Factory.php (hunk 4 above)
Comment (bordoni) [reply src/Tribe/Template_Factory.php]: It's deprecated code.

File: src/Tribe/Template_Factory.php (hunk 3 above)
Comment (bordoni) [reply src/Tribe/Template_Factory.php]: It's deprecated code.

File: src/Tribe/Main.php (hunk 2 above)
Comment (bordoni) [reply src/Tribe/Main.php]: Good catch.

**Comment by bordoni** [comment]
Ready for another look; @mitogh and @barryhughes

File: src/Tribe/Template_Factory.php (hunk 3 above)
Comment (ggwicz) [reply src/Tribe/Template_Factory.php]: does using `_deprecated_function()` make sense here and for these other deprecated asset methods?

File: src/Tribe/Template_Factory.php (hunk 4 above)
Comment (ggwicz) [reply src/Tribe/Template_Factory.php]: does using `_deprecated_function()` make sense here and for these other deprecated asset methods?

File: src/Tribe/Template_Factory.php (hunk 3 above)
Comment (bordoni) [reply src/Tribe/Template_Factory.php]: Nopes, other plugins will use these methods for now. We want to avoid deprecating it with notices for now.

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M18.07*, but this was pull requested against *release/M18.06*.

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M18.08*, but this was pull requested against *release/M18.07*.

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M18.09*, but this was pull requested against *release/M18.08*.

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M18.11*, but this was pull requested against *release/M18.10*.

**Comment by bordoni** [comment]
The common version here is not loading, that is why the tests are failing. Make sure you check on the required versions of `common`, make sure it's submodule is synced corectly.

**Pull request merged**